packages/webpack-cli/lib/groups/basicResolver.js

```diff
@@ -34,7 +71,76 @@ function resolveArgs(args) {
             finalOptions.options[arg] = args[arg];
         }
     });
+    if (outputPath) {
+        finalOptions.options.output = { path: path.resolve(outputPath) };
+    }
+
+    if (stats !== undefined) {
+        finalOptions.options.stats = stats;
+    }
+    if (json) {
+        finalOptions.outputOptions.json = json;
+    }
+
+    if (hot) {
+        const { HotModuleReplacementPlugin } = require('webpack');
+        const hotModuleVal = new HotModuleReplacementPlugin();
+        if (finalOptions.options && finalOptions.options.plugins) {
+            finalOptions.options.plugins.unshift(hotModuleVal);
+        } else {
+            finalOptions.options.plugins = [hotModuleVal];
+        }
+    }
+    if (prefetch) {
+        const { PrefetchPlugin } = require('webpack');
+        const prefetchVal = new PrefetchPlugin(null, args.prefetch);
+        if (finalOptions.options && finalOptions.options.plugins) {
+            finalOptions.options.plugins.unshift(prefetchVal);
+        } else {
+            finalOptions.options.plugins = [prefetchVal];
+        }
+    }
+    if (analyze) {
+        if (packageExists('webpack-bundle-analyzer')) {
+            // eslint-disable-next-line node/no-extraneous-require
+            const { BundleAnalyzerPlugin } = require('webpack-bundle-analyzer');
+            const bundleAnalyzerVal = new BundleAnalyzerPlugin();
+            if (finalOptions.options && finalOptions.options.plugins) {
+                finalOptions.options.plugins.unshift(bundleAnalyzerVal);
+            } else {
+                finalOptions.options.plugins = [bundleAnalyzerVal];
+            }
+        } else {
+            await promptInstallation('webpack-bundle-analyzer', () => {
+                error(`It looks like ${yellow('webpack-bundle-analyzer')} is not installed.`);
+            })
+                .then(() => success(`${yellow('webpack-bundle-analyzer')} was installed sucessfully.`))
+                .catch(() => {
+                    error(`Action Interrupted, Please try once again or install ${yellow('webpack-bundle-analyzer')} manually.`);
+                    process.exit(2);
+                });
+        }
+    }
+    if (target) {
+        finalOptions.options.target = args.target;
+    }
+
+    if (Array.isArray(configOptions)) {
+        // Todo - handle multi config for all flags
+        finalOptions.options = configOptions.map(() => ({ ...finalOptions.options }));
+        configOptions.forEach((configObject, index) => {
+            const computedMode = assignMode(mode, configObject);
+            if (computedMode) {
+                finalOptions.options[index].mode = computedMode;
+            }
+        });
+    } else {
+        const computedMode = assignMode(mode, configOptions);
+        if (computedMode) {
+            finalOptions.options.mode = computedMode;
+        }
+    }
     return finalOptions;
-}
+};
```

Comment:
Let's move it to WebpackCLI classs as method `resolveArgs`

Reply:
No groups anymore

Reply:
yes we can move but then it will be a big file, still continue?

Reply:
I don't think it will be very big, keeping logic in one places sometimes is better than separate it, this will help us in future improvements and refactor to see everything

Reply:
cool lets do it